### DISCUSSION TITLE
Suggestions Box

### DISCUSSION THREAD

**Discussion by RomFR57** [RomFR57/fastfinance#1] [Ideas]
Post here the indicators you want me to add :smile_cat:

**Comment by macordalfy** [comment]
choppiness index
renko charts

**Reply by RomFR57** [reply]
i have added chopiness index, i will add renko charts later when i get time because i'm doing a video game right now

**Reply by macordalfy** [reply]
awesome thanks a lot

**Comment by mupsje** [comment]
Hi, can you please ad COG (centre of Gravity) and
ZLSMA from @veryfid

Other question...
Man I love your work...
But how do I make this into that chart thing like your youtube?
Can you sent me a example ?
Also just learning Python with jupyter-labs ;)
regards Remco

**Reply by RomFR57** [reply]
Hi, I put COG and ZLSMA in the wishlist but I'm into the Renko Charts now, maybe you have to wait a little for them.
In my youtube video, you see my automated crypto bot which use fastfinance and my candlemaker system based on Plotly.
I can share help for the candlemaker script for the ones who ask me by email or wait for me to release the Candlemaker Repository in next weeks.
I use PyCharm, I recommend it because you have package management, link to Git and you can still do some Python if internet is broken.
Thank you for your support

Wishlist : 
1 - Renko Charts
2 - Centre Of Gravity
3 - ZLSMA

**Reply by mupsje** [reply]
Hi, I sent you a email back ;)

**Reply by macordalfy** [reply]
Hi hope all is well on your side, by any chance you managed to add renko charts ?

**Reply by RomFR57** [reply]
Hi, I'm ok and you guys ? 
Sorry for not making new updates, I put programming aside because I struggled with a lot of little problems the lasts months : arm broken, house moving, ex girlfriend and the most important find money. But things go better now.
I tried to code Renko Charts and I didn't manage to make it works because I didn't spend enough time on it. So I'm not even at the step of converting it to Numba. I like Renko Charts but it's an awful algorithm to code as it's not time frame dependant so I need to make a special display to check if it works as I did for Ichimoku Cloud. I will take a look again and try to finish the users wishlist and my secret wishlist.

**Reply by macordalfy** [reply]
thanks for the reply, and so sorry to hear about the arm and all the rest, glad to hear that things got better :) .. thanks in advance for all the work on such an amazing library

**Reply by RomFR57** [reply]
> Hi hope all is well on your side, by any chance you managed to add renko charts ?

Here is my first attempt for Renko Charts, can you try it ?

```
@jit(nopython=True)
def renko(data, brick_size):
    """
    Renko Charts
    :param data: np.ndarray
    :param brick_size: float
    :rtype: np.ndarray, np.ndarray, np.ndarray
    :return: index, up, down
    """
    dif = np.diff(data)
    bricks = np.zeros_like(dif)
    bricks[0] = np.ceil(abs(dif[0]) / brick_size) * np.sign(dif[0])
    for i in range(1, len(dif)):
        b = np.ceil(abs(dif[i]) / brick_size) * np.sign(dif[i])
        bricks[i] = -1 * bricks[i-1] if abs(bricks[i-1] + b) >= 1 else bricks[i-1] + b
    up = np.empty_like(data)
    down = np.empty_like(data)
    c = data[0]
    for i in range(len(bricks)):
        if bricks[i] > 0:
            up[i] = c + brick_size
            down[i] = c
            c += brick_size
        elif bricks[i] < 0:
            up[i] = c
            down[i] = c - brick_size
            c -= brick_size
        else:
            up[i] = c
            down[i] = c
    return np.arange(len(data)) + 1, up, down

```

**Comment by mupsje** [comment]
Hope you are doing great, myself also just released from hospital after several surgery. 
Picking up the trading next month, first reevaluate .

**Reply by RomFR57** [reply]
Hope you are better now, sorry for make you wait a lot guys, I'm about to buy new computer, I'll be active soon, I don't forget you
